examples/denoise_covar2d.py

```diff
@@ -75,23 +76,26 @@
 # Generate 2D clean images from input 3D map. The following statement can be used from the sim object:
 # imgs_clean = sim.clean_images(start=0, num=num_imgs)
 # To be consistent with the Matlab version in the numbers, we need to use the statements as below:
+logger.info('Generate random distributed rotation angles and obtain corresponding 2D clean images.')
 rots = qrand_rots(num_imgs, seed=0)
 imgs_clean = vol2img(vols[..., 0], rots)
 
 # Assign the CTF information and index for each image
 h_idx = np.array([filters.index(f) for f in sim.filters])
 
 # Evaluate CTF in the 8X8 FB basis
-h_ctf_fb = [radial_filter2fb_mat(filt.evaluate_k, ffbbasis) for filt in filters]
+h_ctf_fb = [radial_filter2fb_mat(filt.evaluate, ffbbasis) for filt in filters]
 
 # Apply the CTF to the clean images.
+logger.info('Apply CTF filters to clean images.')
 imgs_ctf_clean = Image(sim.eval_filters(imgs_clean))
 sim.cache(imgs_ctf_clean)
```

Comment:
What does this do? Is it necessary?

Reply:
Currently the simulation class only generates images from Gaussian blob volume. This cache operation will replace the images from Gaussian blob volume with the images from experimental volume. This is only for the consistency with the replaced Gaussian volume in  a previous operation, `sim.vols = vols` in line 71.

Reply:
Ok. That's a bad way to use a simulation. As discussed yesterday, we need a new type of `ImageSource` (simply backed by a stack of images that we provide) for this type of use case. A `cache()` function should never accept a stack of images as input, it should simply cache the *existing* images in that `ImageSource`. Let's make an issue for this.